### PR TITLE
create weather controller

### PR DESCRIPTION
**Pull request by YWAMWeessies**

### CODE REVIEW

File: behind-the-tankard/Assets/Scripts/General/DayController.cs [hunk 1]
```diff
@@ -50,24 +31,21 @@ public class DayStage
...
-        _rain = GameObject.FindGameObjectWithTag("Rain").GetComponent<ParticleSystem>();
-        _currentWeather = (Weather)UnityEngine.Random.Range(0, Enum.GetNames(typeof(Weather)).Length);
-        _thunder = GameObject.FindGameObjectWithTag("Thunder");
-        _thunder.SetActive(false);
+        StartCoroutine(CalculateTime());        
         foreach (DayStage dayStage in dayStages)
         {
             _dayTime += dayStage.duration;
         }
+        _weather = gameObject.GetComponent<WeatherController>();
```
Comment (davidsalsa) [inline behind-the-tankard/Assets/Scripts/General/DayController.cs]: useless empty space?

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            }
+        }
+    }
+
+    public Light GetLight()
+    {
+        return _light;
+    }
+
+    public void SetLightIntensity(float lightIntensity)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: Don't bother with creating getters and setters, that is not C# convention.
Make the variable public instead:
```cs
// This variable has a public getter and a private setter
public Light Light { get; private set; }
// This variable has a custom getter and setter
public float LightIntensity {
    get => Light.lightIntensity;
    set => Light.lightIntensity = value;
}

File: behind-the-tankard/Assets/Scripts/General/DayController.cs [hunk 3]
```diff
@@ -126,71 +109,23 @@ public float GetBusynessFactor()
     {
 
         float _busynessFactor = 0;
```
Comment (davidsalsa) [inline behind-the-tankard/Assets/Scripts/General/DayController.cs]: business*

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+    private float _duration = 5;
+    private Weather _currentWeather;
+    private ParticleSystem _rain;
+    private GameObject _thunder;
+    private bool _thunderCoroutine = false;
+    private const int _sunRainEmmision = 0;
+    private const int _lightingLuckyNumber = 3;
+    private const int _maxLightingChanceVariable = 5;
+    private const float _intervalLightningCheck = 1f;
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: The naming convention for private constants is CamelCase starting with a capital letter.
Alternatively you can make them configurable in the inspector.

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    };
+
+    private float _duration = 5;
+    private Weather _currentWeather;
+    private ParticleSystem _rain;
+    private GameObject _thunder;
+    private bool _thunderCoroutine = false;
+    private const int _sunRainEmmision = 0;
+    private const int _lightingLuckyNumber = 3;
+    private const int _maxLightingChanceVariable = 5;
```
Comment (davidsalsa) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: Thunder?

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    // Update is called once per frame
+    void Update()
+    {
+        ChangeWeather();
+    }
+
+    private void ChangeWeather()
+    {
+
+        float _t = Mathf.PingPong(Time.deltaTime, _duration) / _duration;
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: Please remove the underscore from the local variable name

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                break;
+        }
+    }
+
+    private IEnumerator ActivateLightning()
+    {
+        while (true)
+        {
+            yield return new WaitForSeconds(_intervalLightningCheck);
+            int _chancenumber = UnityEngine.Random.Range(0, _maxLightingChanceVariable);
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: Please remove the underscore from the local variable name

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+            if (_chancenumber == _lightingLuckyNumber)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: A better more intuitive way of deciding a random event is generating a random number between 0 and 100 and checking if its smaller than the threshold:
```cs
// Set a chance of 20%
const float chance = 20;
if (Random.Range(0f, 100f) <= chance)
{
    // do random stuff
}
```

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+    private IEnumerator ActivateLightning()
+    {
+        while (true)
+        {
+            yield return new WaitForSeconds(_intervalLightingCheck);
+            
+            if (UnityEngine.Random.Range(0f, 100f) <= chance)
+            
+                {
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: Please fix the indenting of this bracket

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+                _lightning.SetActive(true);
+                yield return new WaitForSeconds(lightingDuration);
+                _lightning.SetActive(false);
+                yield return null;
+
+            }
+            yield return null;
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: What is the use of the two `yield return null` statements?

File: behind-the-tankard/Assets/Scripts/General/WeatherController.cs [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+        Thunder
+    };
+
+    private float _duration = 5;
+    private Weather _currentWeather;
+    private ParticleSystem _rain;
+    private GameObject _lightning;
+    private bool _lightningCoroutine = false;
+    private int _sunRainEmmision = 0; 
+    private float chance = 20;
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/General/WeatherController.cs]: Rename this variable to be more descriptive and to include an underscore.
Or make it public and configurable from the inspector

**Pull request merged**